### PR TITLE
Add common router. Add tests.

### PR DESCRIPTION
**Pull request by Kun-Kun**
Added router, for now without a factory, but I will implement it after merge all rourers to master branch.
This tests this thread depend on the other routers. In tests present a unit test and test with all handlers and cmd routers.

### CODE REVIEW

File: softgroup/basis/common-router/src/main/java/com/softgroup/common/router/api/AbstractCommandRouterHandler.java [hunk 1]
```diff
@@ -0,0 +1,17 @@
...
+
+import com.softgroup.common.protocol.Request;
+
+/**
+ * Created by user on 26.02.2017.
+ */
+public abstract class AbstractCommandRouterHandler<T extends Handler> extends AbstractRouterHandler<T>{
+
+    @Override
+    public abstract String getName();
```
Comment (Dominys) [inline softgroup/basis/common-router/src/main/java/com/softgroup/common/router/api/AbstractCommandRouterHandler.java]: @Kun-Kun It's already abstract in AbstractRouterHandler or not ?

File: softgroup/basis/common-router/src/main/java/com/softgroup/common/router/api/AbstractRequestHandler.java [hunk 2]
```diff
@@ -6,7 +6,7 @@
 import com.softgroup.common.protocol.Response;
 import com.softgroup.common.protocol.ResponseData;
 
-public class AbstractRequestHandler<T extends RequestData, R extends ResponseData> implements RequestHandler {
+public abstract class AbstractRequestHandler<T extends RequestData, R extends ResponseData> implements RequestHandler {
 	@Override
 	public String getName() {
```
Comment (Dominys) [inline softgroup/basis/common-router/src/main/java/com/softgroup/common/router/api/AbstractRequestHandler.java]: @Kun-Kun Why you need default null return value ?

File: softgroup/services/common/common-impl/pom.xml [hunk 3]
```diff
@@ -0,0 +1,53 @@
...
+         xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance"
+         xsi:schemaLocation="http://maven.apache.org/POM/4.0.0 http://maven.apache.org/xsd/maven-4.0.0.xsd">
+    <parent>
+        <artifactId>common</artifactId>
+        <groupId>com.softgroup</groupId>
+        <version>0.0.1</version>
+    </parent>
+    <modelVersion>4.0.0</modelVersion>
+
+    <artifactId>common-impl</artifactId>
```
Comment (Dominys) [inline softgroup/services/common/common-impl/pom.xml]: @Kun-Kun What is module common-impl O_o ???

File: softgroup/services/common/pom.xml [hunk 4]
```diff
@@ -0,0 +1,58 @@
...
+        <artifactId>services</artifactId>
+        <groupId>com.softgroup</groupId>
+        <version>0.0.1</version>
+    </parent>
+    <modelVersion>4.0.0</modelVersion>
+
+    <artifactId>common</artifactId>
+    <packaging>pom</packaging>
+    <modules>
+        <module>common-impl</module>
```
Comment (Dominys) [inline softgroup/services/common/pom.xml]: @Kun-Kun WTF is common-impl ???

File: softgroup/basis/common-router/src/main/java/com/softgroup/common/router/api/AbstractCommandRouterHandler.java (hunk 1 above)
Comment (Kun-Kun) [inline softgroup/basis/common-router/src/main/java/com/softgroup/common/router/api/AbstractCommandRouterHandler.java]: Yes,it's my mistake

File: softgroup/basis/common-router/src/main/java/com/softgroup/common/router/api/AbstractRequestHandler.java (hunk 2 above)
Comment (Kun-Kun) [inline softgroup/basis/common-router/src/main/java/com/softgroup/common/router/api/AbstractRequestHandler.java]: After making its abstract i can clear this meaningless code (getName, handle), but untill heirs dont realize real handle method i propose return null by default in AbstractRequestHandler.
I aggre to delete getName method.

File: softgroup/services/common/common-impl/pom.xml (hunk 3 above)
Comment (Kun-Kun) [inline softgroup/services/common/common-impl/pom.xml]: Unsuccessful module, consonant with common-router, This module implements the first router that routes by type. I can rename as type-router-impl. Or your suggestion

File: softgroup/services/common/pom.xml (hunk 4 above)
Comment (Kun-Kun) [inline softgroup/services/common/pom.xml]: Unsuccessful module, consonant with common-router, This module implements the first router that routes by type. I can rename as type-router-impl. Or your suggestion

**Pull request merged**